base/doc/ltnews40.tex

```diff
@@ -273,6 +273,19 @@ \section{Improvement to \XeTeX\ \cs{showhyphens}}
 %
 \githubissue{1380}
 
+\section{Avoid code duplication in rollback}
+
+When the kernel uses \cs{AddToHook} in a region that might be 
+rolled back (which happens in a few places) and a document requests a
+rollback, we have the situation that the hook already contains code,
+which is then added once more (or slightly differently) during the
+rollback resulting in code duplication or worse in errors. This has
+now been corrected by dropping the code chunk if it exists prior to
```

Comment:
Not sure if I have correct try interpreted what you are trying to explain here, but a few suggestions that may help others:

rollback, then we have . . . 
contains code that is then added to the . . .
(NOTE: no comma after code above is grammatically important here)
. . . during the rollback; this results in code 
duplication or, worse, in errors. 
. . . by dropping any such code chunk (if there is one) prior to . . .

Reply:
the code it already contains isn't added -- it is already in the hook. Something is added on top which might be the same or it might not

rewritten and the other changes made as suggested